react/features/polls/components/PollChoice.js

```diff
@@ -0,0 +1,188 @@
+// @flow
+
+import React, { Component } from 'react';
+import { FieldTextStateless } from '@atlaskit/field-text';
+
+type Props = {
+
+    /**
+     * Item ID in list.
+     */
+    id: string,
+
+    /**
+     * Container delete method.
+     */
+    deleteHandler?: Function,
+
+    /**
+     * Can be removed, or changed. Need remove and text handlers.
+     */
+    editable: boolean,
+
+    /**
+     * True if item can't be voted.
+     */
+    disabled: boolean,
+
+    /**
+     * Wether this item is selected before by the user or not.
+     */
+    selected: boolean,
+
+    /**
+     * Text of the voting option.
+     */
+    text: string,
+
+    /**
+     * Container text update method.
+     */
+    textChangeHandler?: Function,
+
+    /**
+     * Option voting handling method.
+     */
+    voteHandler?: Function,
+
+    /**
+     * Number of votes for this option.
+     */
+    votes: number
+
+};
+
+/**
+ * Poll option view component.
+ */
+class PollChoice extends Component<Props, *> {
+
+    /**
+     * Constructor.
+     *
+     * @inheritdoc
+     */
+    constructor(props: Props) {
+        super(props);
+
+        this._onClickHandler = this._onClickHandler.bind(this);
+        this._onTextChange = this._onTextChange.bind(this);
+        this._onVoteChange = this._onVoteChange.bind(this);
+    }
+
+    /**
+     * Component render method.
+     *
+     * @inheritdoc
+     */
+    render() {
+        const { disabled, editable, selected } = this.props;
+        const itemControls = this.props.editable
+            ? (
+                <div
+                    className = { 'pollChoiceRemoveButtonContainer' } >
+                    <button
+                        id = { 'pollChoiceRemoveButton' }
+                        onClick = { this._onClickHandler }
+                        type = { 'button' } >
+                        ×
+                    </button>
+                </div>
+            ) : (
+                <div
+                    className = { 'pollChoiceVotesContainer' } >
+                    <label>
+                        { this.props.votes }
+                    </label>
+                </div>
+            );
+
+        return (
+            <li>
+                <div
+                    className = { 'pollChoiceContainer' } >
+                    <div
+                        className = { 'pollVoteButtonContainer' } >
+                        <input
+                            checked = { selected }
+                            disabled = { disabled }
+                            id = { 'pollVoteButton' }
+                            onClick = { this._onVoteChange }
```

Comment:
You might want to be using onChange here instead of onClick. I think the preferred callback, based on react docs, is onChange.

Reply:
Maybe remove it also